tests/test_memory_leak.py

```diff
@@ -156,7 +156,7 @@ def epochs(self) -> int:
 
     @property
     def max_memory_threshold_mb(self) -> float:
-        return 1200
+        return 1700
```

Comment:
Shouldn't we be using less memory now? Or is this unrelated to deduplication of states?

Reply:
This is unrelated. This creeped into the PR from the main PR.